File: app/src/main/java/tech/salroid/filmy/parser/WatchListMovieParseWork.java
package tech.salroid.filmy.parser;

import android.content.Context;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import java.util.ArrayList;
import java.util.List;

import tech.salroid.filmy.data.WatchlistData;

public class WatchListMovieParseWork {

    private String result;

    public WatchListMovieParseWork(Context context, String result) {
        this.result = result;
    }

    public List<WatchlistData> parseWatchList() {
        final List<WatchlistData> favouriteArray = new ArrayList<>();
        WatchlistData favouriteData;

        try {
            JSONObject jsonobject = new JSONObject(result);
            JSONArray jsonArray = jsonobject.getJSONArray("results");

            for (int i = 0; i < jsonArray.length(); i++) {

                String id = jsonArray.getJSONObject(i).getString("id");
                String poster = "http://image.tmdb.org/t/p/w185" + jsonArray.getJSONObject(i).getString("poster_path");
                String title = jsonArray.getJSONObject(i).getString("original_title");
                favouriteData = new WatchlistData(id);

                if (!(poster.equals("null"))) {
                    favouriteData.setId(id);
                    favouriteData.setTitle(title);
                    favouriteData.setPoster(poster);
                    favouriteArray.add(favouriteData);
                }

            }
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return favouriteArray;
    }
}

File: app/src/main/java/tech/salroid/filmy/parser/WatchListMovieParseWork.kt
/*
package tech.salroid.filmy.parser

import org.json.JSONException
import org.json.JSONObject
import tech.salroid.filmy.data.WatchlistData
import java.util.*

class WatchListMovieParseWork(private val result: String) {

    fun parseWatchList(): List<WatchlistData> {
        val favouriteArray: MutableList<WatchlistData> = ArrayList()
        var favouriteData: WatchlistData

        try {
            val jsonObject = JSONObject(result)
            val jsonArray = jsonObject.getJSONArray("results")

            for (i in 0 until jsonArray.length()) {
                val id = jsonArray.getJSONObject(i).getString("id")
                val poster = "http://image.tmdb.org/t/p/w185" + jsonArray.getJSONObject(i)
                    .getString("poster_path")
                val title = jsonArray.getJSONObject(i).getString("original_title")

                favouriteData = WatchlistData(id)
                if (poster != "null") {
                    favouriteData.id = id
                    favouriteData.title = title
                    favouriteData.poster = poster
                    favouriteArray.add(favouriteData)
                }
            }

        } catch (e: JSONException) {
            e.printStackTrace()
        }
        return favouriteArray
    }
}*/
